### PR TITLE
implement sub query filter

### LINKED ISSUE #349: implement sub_query filter

### PR DESCRIPTION
**Pull request by nehahirve**
Fixes #349

### CODE REVIEW

**Comment by richardolsson** [comment]
Christmas eve! 🤶

File: src/locale/misc/smartSearch/en.yml
Comment (richardolsson) [inline src/locale/misc/smartSearch/en.yml:140]: Why are the following under this separate key `query` and not under the `sub_query` key? That seems to be the pattern of other filters.

Comment (richardolsson) [reply src/locale/misc/smartSearch/en.yml:140]: Actually taking a closer look I realize there are many cases where things are not contained under the scope in which they live in the UI. Maybe there is some reason I'm missing, or maybe this is something to handle separately in the refactor issue?

File: src/components/smartSearch/filters/SubQuery/index.tsx
Comment (richardolsson) [inline src/components/smartSearch/filters/SubQuery/index.tsx:1]: Why is this required? Disabling linting rules entirely takes away the benefit of having them in the first place. There can of course be exceptions, but they should be specific for each occasion and not cover entire modules.

I realize taking a closer look at existing files that this has been done before and I haven't caught it. So maybe this is another something to put into the refactor issue?

**Review by richardolsson — COMMENTED** [review]
Looks great and works as expected! I have some comments below before I approve. I think we need to change them, but not necessarily right now. What do you think?

File: src/locale/misc/smartSearch/en.yml
Comment (nehahirve) [reply src/locale/misc/smartSearch/en.yml:140]: This started when I moved timeFrame out of scope; I began to put all similar potential reusable 'snippets' in their own scope, but of course, none of them except timeFrame are actually reused. 
Added to #337

File: src/components/smartSearch/filters/SubQuery/index.tsx
Comment (nehahirve) [reply src/components/smartSearch/filters/SubQuery/index.tsx:1]: Added to #337

File: src/locale/misc/smartSearch/en.yml
Comment (richardolsson) [reply src/locale/misc/smartSearch/en.yml:140]: I think this is a reasonable approach, although I would add them all under a common scope in the YAML, e.g. `misc.smartSearch.widgets` or something. But let's rivisit that for #337.

**Review by richardolsson — APPROVED** [review]
Approving after discussion! Green light to merge. 🙌

**Pull request merged**
